### PR TITLE
Macos support

### LINKED ISSUE #2: OSX support
I've had a look at some of the alternatives of doing a stack walk on stack overflow and I want to try and see if I can use unwind-rs to achieve this.

Is anyone working on OSX support? Also for object loading should we use object crate or findshlibs - to my untrained eye they seem to do similar things.

I'm probably biting off more than I can chew, but rust has made me a bit more adventurous... any help / pointers on what I need to do to get it working on OSX much appreciated!

### PR DESCRIPTION
**Pull request by bjorn3**
Based on #29, #30 and #31.

Fixes #2

I did something wrong causing it to not find the FDE for an address, but I dont know what.

### CODE REVIEW

File: unwind/src/unwind_helper.c [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (bjorn3) [inline unwind/src/unwind_helper.c]: This really broke using `nano`as editor.

File: unwind/src/lib.rs [hunk 2]
```diff
@@ -96,45 +135,65 @@ struct UnwindInfo<R: Reader> {
...
+                    //println!("{:?}", fde);
                 }
             }
+            i += 1;
+            if i > 10 {
+                break;
+            }
         }
 
+        fde = eh_frame.fde_for_address(bases, address, EhFrame::cie_from_offset).unwrap();
```
Comment (bjorn3) [inline unwind/src/lib.rs]: It fails here

Comment (philipc) [reply unwind/src/lib.rs]: This works on linux (after I hack it to use EhRef::WithoutHeader for linux too).

**Comment by philipc** [comment]
While it should be possible to get it to work without, I think you should try to use `eh_frame_hdr` on macos. Some work on that was started [here](https://github.com/gimli-rs/findshlibs/commit/2ae63a38b1d54531eedfeb63d0a48efad1eb3e98).

**Comment by bjorn3** [comment]
`.eh_frame_hdr` doesnt seem to exist on macOS.

> Some work on that was started here.

It would be nice to have that commit on master, as it allows getting `.eh_frame` on macOS without having to copy and leak it and without having to deal with fat libraries.

**Comment by philipc** [comment]
That commit is lacking linux support, so merging it to master is non-trivial. Any work on it is appreciated. Be aware that the `EhFrame` trait will be hard to implement for linux since the section headers aren't available AFAIK.

**Comment by bjorn3** [comment]
I won't work any further on this.

**Pull request closed without merging**